staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -71,14 +88,14 @@ func ShuffleAndDealWithValidation(hashValue uint64, deckSize, handSize int32, pi
 
 // ShuffleAndDealToSlice will use specific pick function to return slices of indices
 // after ShuffleAndDeal
-func ShuffleAndDealToSlice(hashValue uint64, deckSize, handSize int32) []int32 {
+func ShuffleAndDealToSlice(hashValue uint64, deckSize, handSize int32) []int {
```

Comment:
what's the point of s/int32/int/

Reply:
https://github.com/kubernetes/kubernetes/pull/80710#discussion_r312152655

Reply:
> sort.Slice(int32Array, func(i, j int){ return int32Array[i] < int32Array[j]})

we can sort int32 in another style. all the elements in the returned slice are not supposed   to exceed an int32 `deckSize`, returning []int32 is more reasonable here..

Reply:
https://golang.org/ref/spec#Numeric_types promises that an `int` can hold an `int32`.

Yes, we could make clients sort in another way.  But why should we?  Returning `int` makes for simpler client code.

Reply:
On further thought: I understand the dislike of the dissonance between `int32` arguments and `int` returns.  Using `int32` for the arguments is really just inertia from when this was entirely dedicated to the priority and fairness feature, which is configured with `int32` parameters.  But for general purpose use, I think `int` is preferred as it is the natural integer data type.  There is an error here in putting the boundary between priority-and-fairness config fealty and general sensibilities inside this function.  I will recode so that this general purpose function takes `int` size parameters as well as returning `int` cards.